### PR TITLE
Add upgrade documentation

### PR DESCRIPTION
**Pull request by brakthehack**
This commit adds instruction on basic upgrade protocol along
with some version caveats.

### CODE REVIEW

File: docs/upgrade.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+## Prerequisites
+
+- A currently running HAProxy VM
+- A target version greater than the current version
+- The ability to deploy VMs
+
+## Steps
+
+- Confirm you have enough CPU and disk space for the new VM and the old VM.
+- Find and copy `/etc/haproxy/server.crt` and `/etc/haproxy/server.key` out of the instance. Optionally copy the CA files if you need them.
```
Comment (mayankbh) [inline docs/upgrade.md:20]: It should be possible to use a new certificate with the same signing CA, right? (assuming the consumer was configured with said CA)

File: docs/upgrade.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+- If you have made any customizations to `/etc/haproxy/haproxy.cfg` or `/etc/haproxy/dataplaneapi.cfg` then make backups or notes of those customizations at this time.
+- Deploy the new VM with an _identical_ configuration as the currently running instance using the existing HAProxy `server.crt` and `server.key` as inputs.
+- Power down the old instance. You can revert back to this image if something goes wrong with the upgrade.
```
Comment (mayankbh) [inline docs/upgrade.md:23]:
```suggestion
- Power down the old instance. You can revert back to this instance if something goes wrong with the upgrade.
```

(Do we typically refer to VMs as images?)

File: docs/upgrade.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+Verify haproxy and dataplaneapi services have started and are running:
+
+```
+   systemctl status haproxy
+   systemctl status dataplaneapi
+```
+
+Re-apply any custom changes to the `haproxy.cfg` or `dataplaneapi.cfg` files and restart the services.
+
+On the service side, ensure service-type load balancer service IPs are reachable on both the supervisor and Tanzu Clusters.
```
Comment (mayankbh) [inline docs/upgrade.md:44]:
```suggestion
On the service side, ensure service-type load balancer external IPs are reachable on both the supervisor and Tanzu clusters.
```

Should we call out external IPs to distinguish from the ClusterIP addresses of those services?

**Review by mayankbh — APPROVED** [review]

File: docs/upgrade.md (hunk 1 above)
Comment (brakthehack) [reply docs/upgrade.md:20]: Yes.

File: docs/upgrade.md (hunk 2 above)
Comment (brakthehack) [reply docs/upgrade.md:23]: Yeah, definitely could be clearer. Good suggestion.

File: docs/upgrade.md (hunk 3 above)
Comment (brakthehack) [reply docs/upgrade.md:44]: Good idea.

**Pull request merged**